src/parser.cpp

```diff
@@ -562,11 +562,14 @@ job_t *parser_t::job_get(job_id_t id) {
     return NULL;
 }
 
-job_t *parser_t::job_get_from_pid(int pid) {
+job_t *parser_t::job_get_from_pid(pid_t pid) {
     job_iterator_t jobs;
     job_t *job;
+
+    pid_t pgid = getpgid(pid);
+
     while ((job = jobs.next())) {
-        if (job->pgid == pid) return job;
+        if (job->pgid == pgid) return job;
```

Comment:
No, the two processes should get the same PGID. All processes within a job have one job object and one PGID. However, if job control is turned off, multiple jobs can have the same PGID. It's helpful to remember that turning off job control does not disable jobs, just whether they can be backgrounded/foregrounded independently.

However, the test should probably be tightened up here - we need to make sure that the PID is correct as well as the PGID.